### ISSUE TITLE
`jsx-one-expression-per-line`: `ignoreComments`

### ISSUE THREAD

**Issue by otomad** [eslint-stylistic/eslint-stylistic#287]
### Clear and concise description of the problem

Adding a comment at the end of a jsx line can unexpectedly raise `jsx-one-expression-per-line`. Although I don't assume comment in jsx as a common expression.

For example:
```jsx
<Foo>bar</Foo>{/* TODO: This line may be deleted. */}
```

However, the comment `{/* */}` will be treat as a common expression `{}`.

### Suggested solution

Add a `ignoreComments` option to the `jsx-one-expression-per-line` rule.

```json
"@stylistic/jsx/jsx-one-expression-per-line": [<enabled>, { "allow": "none"|"literal"|"single-child", "ignoreComments": boolean }]
```

### Alternative

Or add a new rule `jsx-lines-around-comment` which similar to [`lines-around-comment`](https://eslint.org/docs/latest/rules/lines-around-comment) but in JSX, and also ignore comments in `jsx-one-expression-per-line`.

### Additional context

Although maybe it is better to write as
```jsx
{/* TODO: The next line may be deleted. */}
<Foo>bar</Foo>
```

I am more accustomed to add a comment after lines in JavaScript.
```javascript
const foo = "bar"; // TODO: Something todo.
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by otomad** [comment]
Sorry, I used the wrong configuration.

**Comment by 01taylop** [comment]
The rule from [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react/commit/9f4b2b96d92bf61ae61e8fc88c413331efe6f0da) added a `non-jsx` option "to allow non-JSX children in one line". I wonder if that change would resolve this issue too?

**Comment by otomad** [comment]
> The rule from [eslint-plugin-react](https://github.com/jsx-eslint/eslint-plugin-react/commit/9f4b2b96d92bf61ae61e8fc88c413331efe6f0da) added a `non-jsx` option "to allow non-JSX children in one line". I wonder if that change would resolve this issue too?

Doesn't work.

```json5
"react/jsx-one-expression-per-line": ["error", { allow: "non-jsx" }],
```

`{/* TODO: xxx */}` must be placed on a new line eslint([react/jsx-one-expression-per-line](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/jsx-one-expression-per-line.md))
